include/eigensolve_quda.h

```diff
@@ -237,25 +222,11 @@ namespace quda
        @param[in] blockType Type of caxpy(_U/L) to perform
        @param[in] je End of j index
        @param[in] offset Position of extra vectors in kSpace
-    */
-    void blockRotate(std::vector<ColorSpinorField *> &kSpace, double *array, int rank, const range &i, const range &j,
-                     blockType b_type);
-
-    /**
-       @brief Rotate part of kSpace
-       @param[in/out] kSpace The current Krylov space
-       @param[in] array The complex rotation matrix
-       @param[in] rank row rank of array
-       @param[in] is Start of i index
-       @param[in] ie End of i index
-       @param[in] js Start of j index
-       @param[in] je End of j index
-       @param[in] blockType Type of caxpy(_U/L) to perform
        @param[in] offset Position of extra vectors in kSpace
     */
-
-    void blockRotateComplex(std::vector<ColorSpinorField *> &kSpace, Complex *array, int rank, const range &i,
-                            const range &j, blockType b_type, int offset);
+    template <typename T, typename matrix_t>
+    void blockRotate(std::vector<ColorSpinorField> &kSpace, matrix_t &array, int rank,
+                     const range &i, const range &j, blockType b_type, int offset);
```

Comment:
Is there a general guidance on when to use `std::vector<ColorSpinorField> &` in the intefaces and when to use `vector_ref<ColorSpinorField>` instead? The former obviously owns the fields, but I am asking more from an interface point of view.

Reply:
This will be a long-term evolution, and beyond the scope of this PR....

I'm thinking that we'll want to favour the use of `cvector_ref<ColorSpinorField>` since it will enable multi-RHS functionality with little effort.  The next step here will be in my follow up PR that adds actual multi-RHS Dirac kernels, and I will continue to push in this direction then.

For example, I think that we may want to upgrade all BLAS functions to accept `cvector_ref`<> arguments.  This would better enable us to have multi-RHS solvers everywhere.

Reply:
And I realize I misread the question. 😄 (was thinking it was `cvector_ref` vs `ColorSpinorField&`)

I think in general using `cvector_ref` should be preferred for lower-level functions, e.g., below the level of solvers.  At the solver level, I could imagine we'll want both, and above the solvers I imagine using `std::vector` is the way to go?  🤷‍♀️ 

Again, this can be something that can be long-term evolution, as we add more multi-RHS functionality we will work out the answer.